### ISSUE TITLE
Available commands are unorganized

### ISSUE THREAD

**Issue by expositionrabbit** [Scintillo/eppabasic#51]
This serves well the innoncent children who just want to see what kind of interesting features the language has. However, any child attempting to find something specific, such as some string operation, may find their task quite hard.

I propose a tag-based organization system, which preserves the alphabetical list but also allows innoncent children to see everything tagged with "string", for example, to see what they can do with strings.

Functions could have multiple tags. "Str" could be tagged with "int" and "string" to ensure that both the child looking for string functions and the one wishing to learn about integers see that integers can be converted to strings. Arguably, this might even improve the aspect of discovery - an innoncent child learns about strings and then looks up all functions tagged with "string" to see what they can do.

A good example can be found in CoolBasic. Its organization allows innoncent children to first learn the core functions and then move on to more complicated topics (such as the object system) which would only confuse at that stage. Still, the "all functions in one list" approach shouldn't be completely removed.

**Comment by fergusq** [comment]
I agree, an alphabetically ordined list is handy when you must find documentation for a function thats name I know.

When I only want to research the available functions, it is much better that I see a semantically ordened list.

For example, an innocent child would love to draw a house using EB. However, she does not know which function to use. She can draw borders using Draw -comammands, but has no idea how to fill these polygons with colour.

As she is not an englishwoman, she doesn't know that verb "fill" means coloring. So how can she search the right commands from a list of hundreths of different possibilities?

If there was a semantically ordened list, she could easily look the coloring command under title "Drawing commands".

**Comment by henkkuli** [comment]
Hi!

This feature has actually been partly implemented as can bee seen in our [development branch](https://raw.githubusercontent.com/Scintillo/eppabasic/68d9543e323312a7ef5e63a4c0d4f1e0d261679c/static/manual/fi/commands/abs.md). Currently we just don't have required build script for compiling different kind of index files. This will be taken care soon.

Henrik Lievonen
